src/couch_file.erl

```diff
@@ -524,6 +525,99 @@ handle_info({'EXIT', _, Reason}, Fd) ->
     {stop, Reason, Fd}.
 
 
+%% first try to read header at end, falling back to chunked reads on failure
+find_last_header(Fd, FileSize) ->
+    ok = file:advise(Fd, 0, FileSize, dont_need), % minimize disk cache pollution
```

Comment:
Here's what http://linux.die.net/man/2/posix_fadvise says:

"In kernels before 2.6.18, POSIX_FADV_NOREUSE had the same semantics as POSIX_FADV_WILLNEED. This was probably a bug; since kernel 2.6.18, this flag is a no-op.

POSIX_FADV_DONTNEED attempts to free cached pages associated with the specified region. This is useful, for example, while streaming large files. A program may periodically request the kernel to free cached data that has already been used, so that more useful cached pages are not discarded instead.

Pages that have not yet been written out will be unaffected, so if the application wishes to guarantee that pages will be released, it should call fsync(2) or fdatasync(2) first."

So, it does sounds like I'm using it incorrectly, and that it should presumably be called on the "chunk" after it's been read from if we want to tell the kernel we don't intend to use the data again. I'm not clear about whether the last paragraph is relevant, however, since we're only reading pages and not writing them.

@rnewson originally suggested using this flag, so I'd be interested in getting his feedback as well on this topic.